src/QueryPipeline/QueryPipelineBuilder.cpp

```diff
@@ -323,7 +324,15 @@ std::unique_ptr<QueryPipelineBuilder> QueryPipelineBuilder::joinPipelines(
     right->pipe.dropExtremes();
 
     left->pipe.collected_processors = collected_processors;
-    right->pipe.collected_processors = collected_processors;
+
+    /// Collect the NEW processors for the right pipeline.
+    QueryPipelineProcessorsCollector collector(*right);
+    /// Remember the last step of the right pipeline.
+    ExpressionStep* step = typeid_cast<ExpressionStep*>(right->pipe.processors.back()->getQueryPlanStep());
```

Comment:
why add this?

Reply:
Sorry, can you describle your question?

Reply:
Why take a type check for the right pipeline here?

Reply:
A type checking is done here to prevent unknown exceptions. //Sorry for the late reply.

Reply:
> Why take a type check for the right pipeline here?

A type checking is done here to prevent unknown exceptions. //Sorry for the late reply.